src/assets/js/promo.js

```diff
@@ -0,0 +1,52 @@
+import dayjs from 'dayjs';
+import isSameOrBefore from 'dayjs/plugin/isSameOrBefore';
+import isSameOrAfter from 'dayjs/plugin/isSameOrAfter';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+
+dayjs.extend(isSameOrBefore);
+dayjs.extend(isSameOrAfter);
+dayjs.extend(customParseFormat);
+
+export default () => {
+  return {
+    startDate: '',
+    endDate: '',
+    now: dayjs(),
+    isPromoActive: false,
```

Comment:
this name is a bit deceptive. How about `hasDeclinedPromo: false`

Reply:
Good catch. I will make the changes.

Reply:
Renamed to `hasSeenPromo`